src/datastore/postgres-store.ts

```diff
@@ -1607,6 +1608,129 @@ export class PgDataStore extends (EventEmitter as { new (): DataStoreEventEmitte
     return { results: parsed, total: count };
   }
 
+  async searchHash({
```

Comment:
Especially since we're going for speed, would we want to perform each of these queries in parallel, instead of serially?

Same for the principal search.

Reply:
Hmm. The downside is that only one query needs to succeed in finding a match. Which means that unless the the last query happens to be the only match, all the parallel queries would likely slow down the single query that ends up finding the match. This of course is dependent on how the managed postgres service is configured (for example, if clustering of some kind is setup that might negate this penalty).

Reply:
I think the infra would support many parallel queries by default, so I don't think it would have much of an effect. Also, the client code already uses a connection pool.

> The downside is that only one query needs to succeed in finding a match

Yeah, but you could still use the same prioritization logic you have here to return one result.

I think the way you have it right now is fine, because it's already prioritized in a way that makes sense. Way more people will look for a TXID than a block hash.

Reply:
I think this is really dependent on how the managed pg instance is configured. For example, the pool is currently using the default `10` for max connections, let's say the managed pg instance is clustered to handle 10 parallel queries in a way that doesn't slow any of them down.

In this situation, we'd see performance gains until usage gets to a point where there are often more than 10 parallel queries at a given time. For example, a handful of users doing a search, a handful loading other pages, etc. Then you'd have the 11th and greater queries pending -- slowed down due to the several queries that were not strictly necessary. 

The number of "free" parallel queries might be _much_ higher than 10. My reservations are based on assumptions about the managed pg instance and future usage rates, I could definitely be off on those -- I need to do some research into the available parallel capacity. I'll make them parallel if you feel strongly about it?

Reply:
I don't feel strongly about it! I'm definitely good with keeping it as-is.